hw20/ItemDeck.cpp

```diff
@@ -0,0 +1,77 @@
+#include <fstream>
+#include <string>
+#include <vector>
+#include "ItemDeck.h"
+#include "Item.h"
+
+
+ItemDeck::ItemDeck()
+{
+	//TODO: Move item's database to file in format:
+	// type;name; additional if needed params for the type
+	//For example:
+	//UndeadWeapon;"Holy Grenade";4		//should call UndeadWeapon("Holy Grenade", 4);
+
+    const std::string filename = "items.txt";
+    std::ifstream file(filename);
+    if (!file.is_open()) {
+        m_itemsDataBase =
+        {
+            new Weapon{"The Sword of DOOM", 5},
+            new UndeadWeapon{"Stinky knife", 2},
+            new UndeadWeapon{"Holy grenade", 4},
+            new HolySword{"Platinum sword", 3},
+            new HolySword{"Diamond sword", 4},
+            new HugeRock{"Huge Rock", 5},
+            new HugeRock{"Mountain", 6},
+        };
```

Comment:
I would move it into seperate helper method to not mix default setup and file setup.
It's typical to call that methods 'rollback strategy methods', let's say `rollbackInit` or `defaultInit`